src/Microsoft.Crank.AzureDevOpsWorker/Program.cs

```diff
@@ -344,5 +356,85 @@ private static Task ErrorHandler(ProcessErrorEventArgs args)
         }
 
         private static string LogNow => $"[{DateTime.Now.ToString("hh:mm:ss.fff")}]";
+
+        internal static void MaterializeFiles(JobPayload jobPayload, string workingDirectory)
+        {
+            if (jobPayload?.Files == null || jobPayload.Files.Count == 0)
+            {
+                return;
+            }
+
+            var baseDir = Path.GetFullPath(string.IsNullOrEmpty(workingDirectory)
+                ? Directory.GetCurrentDirectory()
+                : workingDirectory);
+
+            // Ensure base working directory exists
+            Directory.CreateDirectory(baseDir);
+
+            foreach (var file in jobPayload.Files)
+            {
+                var relativePath = file.Key?.Replace('\\', '/');
```

Comment:
I assume this is to make it work on Windows and Linux as both can use `/`, but should we be using `Path.Seperator` instead?